src/librustc_mir/borrow_check/used_muts.rs

```diff
@@ -84,6 +88,8 @@ impl<'visit, 'cx, 'tcx> Visitor<'tcx> for GatherUsedMutsVisitor<'visit, 'cx, 'tc
             );
             self.remove_never_initialized_mut_locals(*into);
         }
+
+        // FIXME: no super_statement?
```

Comment:
I didn't change anything here, but it seems rather suspicious that this does not call `super_terminator` -- in particular since there can be uses of locals inside statements, and this visitor *does* use `visit_local`.

Cc @spastorino @matthewjasper

Reply:
This looks like a bug

Reply:
@matthewjasper I added this now, at the end of this function. I was not sure if beginning or end was appropriate though.

Reply:
The logic in this area is a bit of a mess and is somewhat redundant. The actual test is being able to do some of that clean up without breaking the existing tests.

Reply:
Well, that seems to work.^^